tests/Unit/NumericalAlgorithms/Spectral/Test_Mesh.cpp

```diff
@@ -242,6 +243,29 @@ void test_serialization() noexcept {
               {{Spectral::Quadrature::GaussLobatto, Spectral::Quadrature::Gauss,
                 Spectral::Quadrature::GaussLobatto}}});
 }
+
+void test_option_parsing() {
+  INFO("Option Parsing creation");
+
+  std::array<std::string, 3> bases{"Chebyshev", "Legendre", "FiniteDifference"};
+  std::array<std::string, 2> quadratures{"Gauss", "GaussLobatto"};
+
+  std::stringstream creation_string;
+  for (int i = 0; i < 12; i++) {        // extents
+    for (size_t j = 0; i < 3; i++) {    // bases
+      for (size_t k = 0; k < 2; k++) {  // quadratures
```

Comment:
After the changes to the arrays I suggested above, these can be changed to range-based for loops (`for (const auto basis : bases)`, etc.).  The bases and quadratures can be directly streamed below to get the names into the creation string.

Reply:
I still need the index to get the corresponding enum type though. Could use `std::array<std::pair<std::string, Spectral::Basis>>` with structured bindings and a range-based loop though Im not sure thats easier to read.

Reply:
With Will's suggestion you would just create the mesh as `Mesh<Dim>{num_points, basis, quadrature}`

Reply:
my bad, I didnt see you changed the type of the array.